### ISSUE TITLE
Unexpected behaviors in attribute completions of JSX/TSX

### ISSUE THREAD

**Issue by ramtinabadi** [microsoft/TypeScript#52873]

**Comment by mjbvz** [comment]
I believe the root cause is that typescript returns the completion:

```
style={$1}
```

@andrewbranch Could we return `style={$0}` instead? This should move the cursor into the braces without entering into snippet mode: https://code.visualstudio.com/docs/editor/userdefinedsnippets#_tabstops

**Comment by ramtinabadi** [comment]
@andrewbranch @mjbvz 
The problem is not only about remaining in the snippet mode (even though that IS indeed a problem).
The main problem is the fact that when in the snippet mode, the IntelliSense suggestions are not in focus. So, while typing, you'd expect to press `enter` to select the suggestion but instead add a line break.

**Comment by andrewbranch** [comment]
I’m definitely open to tweaking what TS returns based on feedback, but I think we intentionally return

```
style={$1}$0
```

so you can insert a value in the braces and then tab to get out. This is the same thing we do with import statement completions:

```
// import useSt| -->
import { useState$1 } from "react";$0
```

so the user can add additional named imports and/or skip to the end of what was inserted. I definitely buy that this strategy is less valuable in the JSX attribute case, where the end of the inserted text is only one character beyond where the cursor lands, and that character can be typed over. But it does sound like the bigger issue is that completions don’t function well in snippet mode—I know @gabritto and @MariaSolOs have expressed concern about this issue recently, as we’ve started to return more snippet completions.

**Comment by gabritto** [comment]
I agree with what @andrewbranch said above. Even if we can change this specific type of completion snippet to use only `$0`, for some other completions (e.g. case completions, if we decide to keep the snippets) we'll still run into the same problem due to having other tab stops/placeholders in the snippet. I think completions should as much as possible work the same way in snippet editing mode.

**Comment by mjbvz** [comment]
@ramtinabadi Try setting `"editor.suggest.snippetsPreventQuickSuggestions": false`

@jrieken In snippet mode, should manually triggered suggestions allow you to press `enter` to accept them?

**Comment by ramtinabadi** [comment]
@gabritto My suggestion is to allow the user, in the snippet mode, to select a suggestion by `enter` but use the `tab` to jump to the next placeholder.

This approach allows the user to interact with the suggestions in the snippet mode as usual and follow the placeholders by pressing `tab`. Setting `"editor.suggest.snippetsPreventQuickSuggestions": false` will fix the problem of selecting the first suggestion but will also select a suggestion after pressing tab instead of moving to the next placeholder.

The problem, obviously, is that some users press `tab` to select a suggestion outside the snippet mode. So probably an option to force the `tab` key to jump to the next placeholder might need to be added which makes things a bit complicated.

**Comment by jrieken** [comment]
related discussion/issue is over here: https://github.com/microsoft/vscode/issues/173387

**Comment by RyanCavanaugh** [comment]
Seems like we + VS Code need to work together to figure out the right solution
